src/map.cpp

```diff
@@ -6910,6 +6910,112 @@ void map::restock_fruits( const tripoint &p, int time_since_last_actualize )
     }
 }
 
+void map::produce_sap( const tripoint &p, int time_since_last_actualize )
+{
+    if( time_since_last_actualize <= 0 ) {
+        return;
+    }
+
+    if( ter_at( p ).id != "t_tree_maple_tapped" ) {
```

Comment:
mapdata.cpp entries are useful for quick checking of the features. If you weren't checking it a lot, it would be better to keep it a string id, but you're checking every single tile of a newly loaded map for tapped maple tree.

If you intend to later generalize it to multiple sap types, the entry in mapdata would be unused, but if you want to keep it maple-exclusive for a longer while, it would be nice for it to have an int id.

If you keep it a string id, it may be a good idea to extract it into a `static const std::string tapped_tree( "t_tree_maple_tapped" );` to make sure it isn't constructed every time. I'm not sure if it is required, since the compiler may see the inline string as just `const char *` and build it only once, but if it doesn't, the string building may incur some extra cost for all tiles (not just the tapped tree ones).